packages/eslint-plugin/tests/schemas.test.ts

```diff
@@ -0,0 +1,91 @@
+import type { RuleModule } from '@typescript-eslint/utils/src/ts-eslint';
+import Ajv from 'ajv';
+import type { JSONSchema4 } from 'json-schema';
+
+import eslintPlugin from '../src';
+
+describe("Validating rule's schemas", () => {
+  // These two have defaults which cover multiple arguments that are incompatible
+  const overrideOptions: Record<string, unknown> = {
+    semi: ['never'],
+    'func-call-spacing': ['never'],
+  };
+
+  for (const [ruleName, rule] of Object.entries(eslintPlugin.rules)) {
+    test(`${ruleName} must accept valid arguments`, () => {
+      if (!isValid(rule, overrideOptions[ruleName] ?? rule.defaultOptions)) {
+        throw new Error(
+          `Options failed validation against rule's schema, with errors: ${JSON.stringify(
+            ajv.errors,
+          )}`,
+        );
+      }
+    });
+
+    test(`${ruleName} rejects arbitrary arguments`, () => {
+      if (isValid(rule, [{ 'arbitrary-schemas.test.ts': true }])) {
+        throw new Error(`Options succeeded validation for arbitrary options`);
+      }
+    });
+  }
+});
+
+describe('Individual rule schema validation', () => {
```

Comment:
[Suggestion] Hmm, we normally put rule-specific things in each rule's specific test files. What do you think about moving these individual ones to closer to their rules?

Reply:
Agreed, this makes sense. Have put the schema validation logic in a separate helper, and moved this out to the appropriate test files.